File: library/src/main/java/name/zeno/android/third/rxjava/ValueTransformer.java
package name.zeno.android.third.rxjava;

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/11/29.
 */
public interface ValueTransformer<Upstream, Downstream>
{
  Downstream apply(Upstream upstream);
}

File: library/src/main/java/name/zeno/android/third/rxjava/ValueTransformer.kt
package name.zeno.android.third.rxjava

/**
 * @author 陈治谋 (513500085@qq.com)
 * @since 2016/11/29.
 */
interface ValueTransformer<in Upstream, out Downstream> {
  fun apply(upstream: Upstream): Downstream
}
